src/frontend/ocamlmerlin/old/old_IO.ml

```diff
@@ -314,8 +314,13 @@ let make_json ?(on_read=ignore) ~input ~output () =
       read buf len
   in
   let lexbuf  = Lexing.from_function read in
-  let input   = Yojson.Basic.(stream_from_lexbuf (init_lexer ()) lexbuf) in
-  let input () = try Some (Stream.next input) with Stream.Failure -> None in
+  let input   = ref (Yojson.Basic.(seq_from_lexbuf (init_lexer ()) lexbuf)) in
+  let input () = match Seq.uncons !input with
+    | Some (hd, tail) ->
+        input := tail;
+        Some hd
+    | None -> None
+  in
```

Comment:
Can use `Seq.to_dispenser` instead.

Reply:
Yes, that's true, it actually obsoletes the entire function. Fixed.